plugins/modules/nd_backup.py

```diff
@@ -0,0 +1,164 @@
+    default: backup
+extends_documentation_fragment: cisco.nd.modules
+"""
+
+EXAMPLES = r"""
+- name: Create a Backup
+  cisco.nd.nd_backup:
+    name: nexus
+    encryption_key: testtest
+    file_location: ./nexus.tgz
+    state: backup
+
+- name: Query a Backup job
+  cisco.nd.nd_backup:
+    name: nexus
+    state: query
+  register: query_result
+
+- name: Query all the backup jobs
+  cisco.nd.nd_backup:
+    state: query
+  register: query_results
+
+- name: Delete a Backup job
+  cisco.nd.nd_backup:
+    name: nexus
+    state: absent
+"""
+
+RETURN = r"""
+"""
+
+from ansible.module_utils._text import to_bytes
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.nd.plugins.module_utils.nd import NDModule, nd_argument_spec, write_file
+
+
+def main():
+    argument_spec = nd_argument_spec()
+    argument_spec.update(
+        name=dict(type="str", aliases=["backup_name"]),
+        encryption_key=dict(type="str", no_log=False),
+        file_location=dict(type="str"),
+        backup_key=dict(type="str", no_log=False),
+        state=dict(type="str", default="backup", choices=["backup", "query", "absent"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "backup", ["name", "encryption_key", "file_location"]],
+            ["state", "absent", ["name"]],
+        ],
+    )
+
+    nd = NDModule(module)
+
+    name = nd.params.get("name")
+    encryption_key = nd.params.get("encryption_key")
+    backup_key = nd.params.get("backup_key")
+    file_location = nd.params.get("file_location")
+    state = nd.params.get("state")
+
+    if encryption_key is not None and len(encryption_key) < 8:
+        nd.fail_json("Please provide a minimum of 8 characters for the encryption key.")
+
+    path = "/nexus/infra/api/platform/v1/exports"
+    # The below path for GET operation is to be replaced by an official documented API endpoint once it becomes available.
+    backup_objs = nd.query_obj("/api/config/class/exports")
+    backups_info = []
+    if name:
+        backups_info = [file_dict for file_dict in backup_objs if file_dict.get("description") == name]
+        if len(backups_info) > 1 and backup_key is None and encryption_key is None:
+            nd.fail_json("Multiple backups with the name '{0}' found. Please provide a backup key for the corresponding backup.".format(name))
+        else:
+            backup_keys = [file_dict.get("key") for file_dict in backups_info]
+            if len(backups_info) == 1:
+                backup_key = backup_keys[0]
+            elif backup_key is not None and backup_key not in backup_keys:
+                nd.fail_json(
+                    "Provided key for the backup '{0}' not found."
+                    " Please provide a valid backup key by querying all the backups and looking up the desired backup key.".format(name)
+                )
```

Comment:
Don't think we need backup_keys assignment and nested conditionals here

```suggestion
        if len(backups_info) > 1 and backup_key is None and encryption_key is None:
            nd.fail_json("Multiple backups with the name '{0}' found. Please provide a backup key for the corresponding backup.".format(name))
        elif len(backups_info) == 1:
            backup_key = backups_info[0].get("key")
        elif backup_key is not None and backup_key not in [file_dict.get("key") for file_dict in backups_info]:
            nd.fail_json(
                "Provided key for the backup '{0}' not found."
                " Please provide a valid backup key by querying all the backups and looking up the desired backup key.".format(name)
            )
```

I don't know the logic of this backup key, but am wondering if for the second statement in the conditional we should check that backup_key is matching backups_info[0].get("key")?